### ISSUE TITLE
[FEATURE REQUEST] - Inlcude multiple data entities for a single service

### ISSUE THREAD

**Issue by go4cas** [ice-services/moleculer-db#2]
At present moleculer-db allows only for a single data model per service. There might be use cases where a service needs to manage multiple data entities (tables in a data schema).

Example:
Service: **Access Control Service** - _a service managing role based access control for the stack_.
Data Entities:
- **Roles**: Access roles (e.g. administrator, user, super user, etc)
- **Role Groups**: Grouping access roles together (e.g. finance, technical, marketing, etc)
- **Functions**: System functions that users might have access to (e.g. edit Customer, delete account, etc)
- **Role Functions**: Mapping Roles to Functions

Currently, I would have to create a service for each data entity, i.e. `role-service`, `roleGroup-service`, `function-service`, `roleRunction-service`.

Each of these services will then link to the underlying data table, using the `model` property of the service, e.g.:
```javascript
model: {
    name: "roles",
    define: {
        _id: Sequelize.INTEGER,
        title: Sequelize.STRING,
        status: Sequelize.STRING
    }
}
```

Accessing the model methods:
```javascript
this.model.findById(ctx.params.id)
```

It would be great if the `model` property could accept an array of model objects. Accessing these from the service, would look something like:
```javascript
this.model.roles.findById(ctx.params.id)
```

**Comment by mradkov** [comment]
+1 Yeah, this is a key feature, that needs to be added.

**Comment by Hexenon** [comment]
is it still open?

**Comment by icebob** [comment]
Yes, but it won't be implemented in this version of moleculer-db.

**Comment by tobydeh** [comment]
Im happy to take a look at implementing this. I don't think it's too difficult but we would need to decide how to change the api for update / find / delete etc.

@icebob do you have any ideas on how the interface should work?

**Comment by b0bben** [comment]
Any news on this?

**Comment by d0whc3r** [comment]
any workaround? do I need to create 1 service per table and communicate with each other?

**Comment by JS-AK** [comment]
with modules
moleculer-db
moleculer-db-adapter-mongoose

u can make one service with multiple tables like this

const DbService = require("moleculer-db");
const MongooseAdapter = require("moleculer-db-adapter-mongoose");
const { MONGO_DB_URI } = process.env

const User = require("../model/user.model");
const Post = require("../model/post.model");

/*
...
*/


module.exports = {
  adapter: new MongooseAdapter(MONGO_DB_URI),
  mixins: [DbService],
  model: {
    Post,
    User,
  },
  
  
  // and handle in this service in customHandle action
  
  actions: {
    customHandle: {
      async handler(ctx) {
        const { User, Deal } = this.adapter.model;
        const { id } = ctx.meta.user;
        const users = await User.find();
        const posts= await Deal.find();
        return { users , posts};
    }
  }
}

**Comment by slipstream00** [comment]
> with modules moleculer-db moleculer-db-adapter-mongoose
> 
> u can make one service with multiple tables like this
> 
> ```
> const DbService = require("moleculer-db");
> const MongooseAdapter = require("moleculer-db-adapter-mongoose");
> const { MONGO_DB_URI } = process.env
> 
> const User = require("../model/user.model");
> const Post = require("../model/post.model");
> 
> /*
> ...
> */
> 
> 
> module.exports = {
>   adapter: new MongooseAdapter(MONGO_DB_URI),
>   mixins: [DbService],
>   model: {
>     User,
>     Post,
>   },
>   
>   
>   // and handle in this service in customHandle action
>   
>   actions: {
>     customHandle: {
>       async handler(ctx) {
>         const { User, Post } = this.adapter.model;
>         const [users, posts] = await Promise.all([User.find(), Post.find()]
>         return { users, posts };
>     }
>   }
> }
> ```

Can we have a sneak peek with what's inside: 
`const User = require("../model/user.model");`

I'm having the same problem.

**Comment by truesteps** [comment]
Is it possible with the sequelize adapter? Maybe show us what's inside the `.model` file?

**Comment by JS-AK** [comment]
> Can we have a sneak peek with what's inside: `const User = require("../model/user.model");`
> 
> I'm having the same problem. @JS-AK

const mongoose = require("mongoose");

const model = new BaseMongooseModel({
	email: {type: String},
	firstName: {type: String},
}, {timestamps: true});

const schema = model.getMongooseSchema();

module.exports = mongoose.model("User", schema);
